backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala

```diff
@@ -42,357 +52,362 @@ class VeloxStringFunctionsSuite extends WholeStageTransformerSuite {
   }
 
   test("ucase") {
     runQueryAndCompare(s"select l_orderkey, ucase(l_comment) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
-    runQueryAndCompare(s"select l_orderkey, ucase(null) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+    runQueryAndCompare(s"select l_orderkey, ucase($NULL_STR_COL) " +
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
   }
 
   ignore("locate") {
     runQueryAndCompare(s"select l_orderkey, locate(l_comment, 'a', 1) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
-    runQueryAndCompare(s"select l_orderkey, locate(null, 'a', 1) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+    runQueryAndCompare(s"select l_orderkey, locate($NULL_STR_COL, 'a', 1) " +
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
   }
 
   test("trim") {
     runQueryAndCompare(s"select l_orderkey, trim('    SparkSQL   ') " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
-    runQueryAndCompare(s"select l_orderkey, trim(null) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+    runQueryAndCompare(s"select l_orderkey, trim($NULL_STR_COL) " +
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
     runQueryAndCompare(s"select l_orderkey, trim(l_comment) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
   }
 
-  test("ltrim/rtrim") {
+  ignore("ltrim/rtrim(trimStr, str)") {
     runQueryAndCompare(s"select l_orderkey, ltrim('SparkSQL   ', 'Spark') " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
     runQueryAndCompare(s"select l_orderkey, ltrim('    SparkSQL   ', 'Spark') " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+      s"from $LINEITEM_TABLE limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
+  }
+
+  test("ltrim/rtrim(str)") {
     runQueryAndCompare(s"select l_orderkey, ltrim('    SparkSQL   ') " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
-    runQueryAndCompare(s"select l_orderkey, ltrim(null) " +
-      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
```

Comment:
Found an unsupported function.

Velox currently only has presto implementation of rtrim, no spark implementation. Only `ltrim/rtrim(str)` is supported, not `ltrim/rtrim(trimStr, str)`.

https://facebookincubator.github.io/velox/functions/presto/string.html#rtrim

Reply:
Exactly. Only default trim string (i.e., space character) is supported by now. Let's ignore the tests where other trim strings are provided, as you did.

Could you please update the below doc to tell user this limitation in `Restrictions` col? Thanks!
https://github.com/oap-project/gluten/blob/main/docs/SupportProgress.md

Reply:
Updated docs for `ltrim`, `rtrim` and `trim`

https://github.com/izchen/gluten/blob/fix_VeloxStringFunctionsSuite/docs/SupportProgress.md